modules/oe_whitelabel_paragraphs/oe_whitelabel_paragraphs.module

```diff
@@ -460,6 +462,79 @@ function oe_whitelabel_preprocess_paragraph__oe_facts_figures(array &$variables)
   }
 }
 
+/**
+ * Implements hook_preprocess_paragraph() for oe_carousel paragraph.
+ */
+function oe_whitelabel_preprocess_paragraph__oe_carousel(array &$variables): void {
+  /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+  $paragraph = $variables['paragraph'];
+  $variables['items'] = [];
+  $cacheability = CacheableMetadata::createFromRenderArray($variables);
+
+  /** @var \Drupal\paragraphs\Entity\Paragraph $sub_paragraph */
+  foreach ($paragraph->get('field_oe_carousel_items')->referencedEntities() as $sub_paragraph) {
+    // Get sub-paragraph translation.
+    $sub_paragraph = \Drupal::service('entity.repository')
+      ->getTranslationFromContext($sub_paragraph, $paragraph->language()->getId());
+    /** @var \Drupal\media\Entity\Media $media */
+    $media = $sub_paragraph->get('field_oe_media')->entity;
+    if (!$media instanceof MediaInterface) {
+      // The media entity is not available anymore, skip the item.
+      continue;
+    }
+    // Retrieve the correct media translation.
+    $media = \Drupal::service('entity.repository')->getTranslationFromContext($media, $paragraph->language()->getId());
```

Comment:
`\Drupal::service('entity.repository')` could/should be moved out of the loop in this case.